Modify rule S6437: Add Docker (APPSEC-462) (#1574)
SonarSource/rspec: rules/S6437/docker/description.adoc

A hard-coded secret has been found in your Dockerfile or Docker image. You should quickly
list where this secret is used, revoke it, and then change it in every system

> that uses it.

Reviewer: This paragraph still has to be reworded to take out the "you" form.

Reviewer 2: missed this one sorry

A hard-coded secret has been found in your Dockerfile or Docker image. You should quickly
list where this secret is used, revoke it, and then change it in every system
that uses it.

If you allow third parties to get a hold of such secret, they
can impersonate legitimate identities and undermine trust within the
organization. +

> It does not matter if the impersonation is malicious: In either case, it is a

Reviewer: > It does not matter if the impersonation is malicious: In either case

I don't think we need to consider the case of impersonation being innocent. Noting that it allows for a breach of the system, leading to possibly catastrophic consequences should be enough.

clear breach of trust in the system, as the systems involved falsely assume
that the authenticated entity is who it claims to be. +
The consequences can be catastrophic.

> Hard-coded secrets or secrets passed using build-time variables to propagate
> configuration entries that contain secrets causes a security risk. The secret

Reviewer: ```suggestion
Both secrets hard-coded in Dockerfiles and secrets passed through as
build-time variables will cause security risks. The secret
```
This suggestion is a bit more readable, but perhaps a bit less precise. Change it as you see fit.

It probably also makes sense to put this earlier in the text.

> information can happen either via Dockerfile disclosure, in the image metadata

Reviewer: "happen" doesn't seem like the right word here. Maybe "can leak" or "can be exposed"?
